### ISSUE TITLE
After e.fit(X) fails, it cannot be called again

### ISSUE THREAD

**Issue by matsui528** [matsui528/rii#7]
Suppose `e.fit(X)` fails, for example the number of vectors is less than 256:
```
~/usr/src/anaconda/anaconda3/envs/sandbox/lib/python3.6/site-packages/nanopq/pq.py in fit(self, vecs, iter, seed)                                          
     61         assert vecs.ndim == 2 
     62         N, D = vecs.shape     
---> 63         assert self.Ks < N, "the number of training vector should be more than Ks"                                                                 
     64         assert D % self.M == 0, "input dimension must be dividable by M"                                                                           
     65         self.Ds = int(D / self.M)                                    

AssertionError: the number of training vector should be more than Ks   
```
Here, even though `e.fit(X)` fails, an empty codec is assigned to `self.fine_quantizer`.  This prohibits us to call `e.fig(X)` again: 

```
~/usr/src/anaconda/anaconda3/envs/sandbox/lib/python3.6/site-packages/rii/rii.py in fit(self, vecs, iter, seed)                                            
    130                               
    131         """                   
--> 132         assert self.fine_quantizer is None, "`fit` should be called only once"                                                                     
    133         assert vecs.dtype == np.float32                              
    134                               

AssertionError: `fit` should be called only once      
```

This isn't a proper behaviour. After a user finds that `e.fit(X)` fails,  she will call `e.fit(X)` again with some new `X`, and it should work. 

This can be solved by assigning an empty codec to `self.fine_quantizer` in the constructor of the Rii class, and replacing an error checking from `assert self.fine_quantier is None` to `assert self.codewords`

**Comment by matsui528** [comment]
Or this can be solved by chaining `fit` function so that `e.fine_quantizer` is not set when `fit` fails?
```
self.fine_quantizer = nanopq.PQ(M=self.M, Ks=self.Ks, verbose=self.verbose).fit(vecs=vecs, iter=iter, seed=seed)
```

**Comment by matsui528** [comment]
This issue was solved in #9 by explicitly defining a codec outside the Rii class
